gen_tests.sh

```diff
@@ -4,66 +4,86 @@ set -e
-  --java_out=../wire-runtime/src/test/proto-java \
-  --registry_class=com.squareup.wire.protos.person.EmptyRegistry \
-  person.proto
+rm -r temp_gen_tests
+mkdir temp_gen_tests
+cp wire-runtime/src/test/proto/person.proto temp_gen_tests/
 
-java -jar ../wire-compiler/target/wire-compiler-*-SNAPSHOT-jar-with-dependencies.jar \
-  --proto_path=../wire-runtime/src/test/proto \
-  --java_out=../wire-runtime/src/test/proto-java \
-  --registry_class=com.squareup.wire.protos.ProtoRegistry \
-  simple_message.proto external_message.proto foreign.proto
+java -jar wire-compiler/target/wire-compiler-*-SNAPSHOT-jar-with-dependencies.jar \
+  --proto_path=temp_gen_tests \
+  --java_out=wire-runtime/src/test/proto-java \
+  --registry_class=com.squareup.wire.protos.person.EmptyRegistry
 
-java -jar ../wire-compiler/target/wire-compiler-*-SNAPSHOT-jar-with-dependencies.jar \
-  --proto_path=../wire-runtime/src/test/proto \
-  --java_out=../wire-runtime/src/test/proto-java \
-  --registry_class=com.squareup.wire.protos.one_extension.OneExtensionRegistry \
-  one_extension.proto
+rm -r temp_gen_tests
+mkdir temp_gen_tests
+cp wire-runtime/src/test/proto/simple_message.proto temp_gen_tests/
+cp wire-runtime/src/test/proto/external_message.proto temp_gen_tests/
+cp wire-runtime/src/test/proto/foreign.proto temp_gen_tests/
+cp -R wire-runtime/src/test/proto/google temp_gen_tests/
+
+java -jar wire-compiler/target/wire-compiler-*-SNAPSHOT-jar-with-dependencies.jar \
+  --proto_path=temp_gen_tests \
+  --java_out=wire-runtime/src/test/proto-java \
+  --registry_class=com.squareup.wire.protos.ProtoRegistry
+
+rm -r temp_gen_tests
+mkdir temp_gen_tests
+cp wire-runtime/src/test/proto/one_extension.proto temp_gen_tests/
+
+java -jar wire-compiler/target/wire-compiler-*-SNAPSHOT-jar-with-dependencies.jar \
+  --proto_path=temp_gen_tests \
+  --java_out=wire-runtime/src/test/proto-java \
+  --registry_class=com.squareup.wire.protos.one_extension.OneExtensionRegistry
 
 # ANDROID
 
-java -jar ../wire-compiler/target/wire-compiler-*-SNAPSHOT-jar-with-dependencies.jar \
-  --proto_path=../wire-runtime/src/test/proto \
-  --java_out=../wire-runtime/src/test/proto-java.android \
-  --android \
-  person.proto
+rm -r temp_gen_tests
+mkdir temp_gen_tests
+cp wire-runtime/src/test/proto/person.proto temp_gen_tests/
+
+java -jar wire-compiler/target/wire-compiler-*-SNAPSHOT-jar-with-dependencies.jar \
+  --proto_path=temp_gen_tests \
+  --java_out=wire-runtime/src/test/proto-java.android \
+  --android
+
+cp wire-runtime/src/test/proto-java.android/com/squareup/wire/protos/person/Person.java \
+   wire-runtime/src/test/proto-java/com/squareup/wire/protos/person/Person.java.android
 
-cp src/test/proto-java.android/com/squareup/wire/protos/person/Person.java \
-   src/test/proto-java/com/squareup/wire/protos/person/Person.java.android
+rm -r wire-runtime/src/test/proto-java.android
 
-rm -r src/test/proto-java.android
+rm -r temp_gen_tests
```

Comment:
Or we probably want to switch to a more partitioned integration test model where we just have multiple proto folders and check in their expected outputs.